go/libraries/doltcore/sqle/index/dolt_index.go

```diff
@@ -748,6 +748,12 @@ func pruneEmptyRanges(sqlRanges []sql.Range) (pruned []sql.Range, err error) {
 				break
 			}
 		}
+		for _, ce := range sr {
+			if lb, ok := ce.LowerBound.(sql.Below); ok && lb.Key == nil {
+				empty = true
+				break
+			}
+		}
```

Comment:
Where is a `LowerBound: sql.Below{Key: nil}` coming from? Shouldn't we be avoiding that?

Reply:
Yeah this is a meaningless range that is produced as a side-effect of my range building refactors in specific edge cases with Left/Right joins. It was just ignored before, because of the way we were searching for cursor items, and panicked with the new direct cursor lookup. The correct fix is probably making this an empty range on the GMS side. I hope to clean this up in a follow-on PR.